bundles/org.openhab.binding.electroluxappliances/src/main/java/org/openhab/binding/electroluxappliances/internal/api/ElectroluxGroupAPI.java

```diff
@@ -0,0 +1,321 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.electroluxappliances.internal.api;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.electroluxappliances.internal.ElectroluxAppliancesBridgeConfiguration;
+import org.openhab.binding.electroluxappliances.internal.ElectroluxAppliancesException;
+import org.openhab.binding.electroluxappliances.internal.dto.AirPurifierStateDTO;
+import org.openhab.binding.electroluxappliances.internal.dto.ApplianceDTO;
+import org.openhab.binding.electroluxappliances.internal.dto.ApplianceInfoDTO;
+import org.openhab.binding.electroluxappliances.internal.dto.ApplianceStateDTO;
+import org.openhab.binding.electroluxappliances.internal.dto.WashingMachineStateDTO;
+import org.openhab.binding.electroluxappliances.internal.listener.TokenUpdateListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link ElectroluxGroupAPI} class defines the Elextrolux Group API
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class ElectroluxGroupAPI {
```

Comment:
Generic comment on this class. 
It currently only uses one type of exception `ElectroluxAppliancesException` that makes it impossible for the user/caller of this class to distinquish erros between a transient network issue (timeout or whatever) and a configuration issue (like authentication, authorisation, token related). In this case case the Thinghandler is the caller and that class could use this to set the different Thing state / details accordingly.

Reply:
OK; I will not change this for now.